resources/asciidoctor/lib/copy_images/extension.rb

```diff
@@ -0,0 +1,69 @@
+require 'csv'
+require 'fileutils'
+require 'set'
+require_relative '../scaffold.rb'
+
+include Asciidoctor
+
+##
+# Copies images that are referenced into the same directory as the output files.
+#
+class CopyImages < TreeProcessorScaffold
+  include Logging
+
+  def initialize name
+    super
+    @copied = Set[]
+  end
+
+  def process_block block
+    return unless block.context == :image
+    uri = block.image_uri(block.attr 'target')
+    return if Helpers.uriish? uri       # Skip external images
+    return unless @copied.add? uri      # Skip images we've copied before
+    source = find_source block, uri
+    return unless source                # Skip images we can't find
+    logger.info message_with_context "copying #{uri}", :source_location => block.source_location
```

Comment:
I'm not sure if info is right here but it doesn't seem too bad. We don't have too many images.